lib/ansible/module_utils/network/eos/facts/l2_interfaces/l2_interfaces.py

```diff
@@ -0,0 +1,99 @@
+from ansible.module_utils.network.eos.argspec.l2_interfaces.l2_interfaces import L2_interfacesArgs
+
+
+class L2_interfacesFacts(object):
+    """ The eos l2_interfaces fact class
+    """
+
+    def __init__(self, module, subspec='config', options='options'):
+        self._module = module
+        self.argument_spec = L2_interfacesArgs.argument_spec
+        spec = deepcopy(self.argument_spec)
+        if subspec:
+            if options:
+                facts_argument_spec = spec[subspec][options]
+            else:
+                facts_argument_spec = spec[subspec]
+        else:
+            facts_argument_spec = spec
+
+        self.generated_spec = utils.generate_dict(facts_argument_spec)
+
+    def populate_facts(self, connection, ansible_facts, data=None):
+        """ Populate the facts for l2_interfaces
+
+        :param module: the module instance
+        :param connection: the device connection
+        :param data: previously collected conf
+        :rtype: dictionary
+        :returns: facts
+        """
+        if not data:
+            data = connection.get('show running-config | section ^interface')
+
+        # operate on a collection of resource x
+        config = data.split('interface ')
+        objs = []
+        for conf in config:
+            if conf:
+                obj = self.render_config(self.generated_spec, conf)
+                if obj:
+                    objs.append(obj)
+        facts = {}
+        if objs:
+            facts['l2_interfaces'] = objs
+        ansible_facts['ansible_network_resources'].update(facts)
+        return ansible_facts
+
+    def render_config(self, spec, conf):
+        """
+        Render config as dictionary structure and delete keys from spec for null values
+
+        :param spec: The facts tree, generated from the argspec
+        :param conf: The configuration
+        :rtype: dictionary
+        :returns: The generated config
+        """
+        config = deepcopy(spec)
+
+        # populate the facts from the configuration
+        config['name'] = re.match(r'(\S+)', conf).group(1).replace('"', '')
+
+        has_access = re.search(r"switchport access vlan (\d+)", conf)
+        if has_access:
+            config["access"] = {"vlan": int(has_access.group(1))}
+
+        has_trunk = re.findall(r"switchport trunk (.+)", conf)
+        if has_trunk:
+            trunk = {}
+            for match in has_trunk:
+                has_native = re.match(r"native vlan (\d+)", match)
+                if has_native:
+                    trunk["native_vlan"] = int(has_native.group(1))
+                    continue
+
+                has_allowed = re.match(r"allowed vlan (\S+)", match)
+                if has_allowed:
+                    # TODO: listify?
+                    trunk["trunk_allowed_vlans"] = has_allowed.group(1)
+                    continue
+            config['trunk'] = trunk
+
+        return utils.remove_empties(config)
```

Comment:
The logic does not validate if the facts are rendered in argspec format. Refer https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/network/ios/facts/interfaces/interfaces.py#L65